sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -571,7 +571,7 @@ long flush(
                             failedRow, error.getRowIndexToErrorMessage().get(failedIndex)),
                         timestamp);
                   } catch (InvalidProtocolBufferException e) {
-                    LOG.error("Failed to insert row and could not parse the result!");
+                    LOG.error("Failed to insert row and could not parse the result!", e);
```

Comment:
I think both of these need to rethrow the exception

Reply:
agreed

Reply:
Do we need to warn the potential data loss? Can we add some actionable steps for users such as validate the data schema matches the given one?

Reply:
If throw exception and not caught in the DoFn then it will retry indefinitely and the pipeline essentially stuck. If choose to re-throw, it should still be handled by the DoFn that calls it.

Reply:
this is the dead-letter codepath. These rows have already been successfully converted to protobuf upstream (by Beam), so failing conversion here is quite unexpected. The only scenario I can think of if there is data corruption on the wire, in which case there's not much we can do.

Reply:
Thanks for the context @reuvenlax. Probably having the exception spit out would help in understanding the context, so this change should be still worth pushing through, what do you think?

All we know today is that it's a `InvalidProtocolBufferException`, so I'm hoping to get more information (we could likely ask the user who got in this situation to use a snapshot, as they have been responsive).

I will try to reproduce this locally.